pkg/extractor/extractor.go

```diff
@@ -0,0 +1,152 @@
+package extractor
+
+import (
+	"github.com/astralkn/k8s-logs-extractor/pkg/kube"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type Extractor interface {
+	Extract(acc *kube.Accessor, outputDir string) error
+}
+
+type PodExtractor struct {
+}
+
+func (_ PodExtractor) Extract(acc *kube.Accessor, outputDir string) error {
+	_, err := acc.DumpInfo(outputDir, "all")
+
+	return err
+}
+
+type CMExtractor struct {
+}
+
+func (_ CMExtractor) Extract(acc *kube.Accessor, outputDir string) error {
```

Comment:
It seems to me like there is a lot of duplicated code for things that could be passed as parameters to the function. 
Instead of doing an extract for all the Different types, consider creating a single extract method, and use the structs to store the data you need.